### PR TITLE
add MiMa settings and test them

### PR DESCRIPTION
**Pull request by scottsand-db**
- run `mimaReportBinaryIssues`command inside `standalone` project to get binary compatibility issues
- e.g. `build/sbt 'standalone/mimaReportBinaryIssues'`
- will be run as part of standalone tests

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -32,3 +32,4 @@ addSbtPlugin("com.jsuereth" % "sbt-pgp" % "1.0.0")
 
 addSbtPlugin("org.xerial.sbt" % "sbt-sonatype" % "2.3")
 
+addSbtPlugin("com.typesafe" % "sbt-mima-plugin" % "0.3.0")
```
Comment (zsxwing) [inline project/plugins.sbt:35]: nit: missing an empty line

File: project/StandaloneMimaExcludes.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * The list of Mima errors to exclude in the Standalone project.
+ */
+object StandaloneMimaExcludes {
+  val ignoredABIProblems = Seq(
+    // Ignore changes to internal Scala codes
+    ProblemFilters.exclude[Problem]("io.delta.standalone.internal.*"),
+
+    // Public API changes in 0.2.0
```
Comment (zsxwing) [inline project/StandaloneMimaExcludes.scala:28]: 0.2.0 -> 0.3.0

**Review by zsxwing — APPROVED** [review]
Thanks! Left some nits.

**Comment by scottsand-db** [comment]
@zsxwing I am unable to merge this btw

**Pull request merged**